engines/engine.h

```diff
@@ -285,128 +373,152 @@ class Engine {
+	 *
+	 * This is to avoid loading a savegame from
 	 * inside the menu loop which causes bugs like #2822778.
 	 *
-	 * @param slot	the slot from which a savestate should be loaded.
+	 * @param slot	The slot from which a save state should be loaded.
 	 */
 	void setGameToLoadSlot(int slot);
 
 	/**
-	 * Indicates whether a game state can be loaded.
+	 * Indicate whether a game state can be loaded.
 	 */
 	virtual bool canLoadGameStateCurrently();
 
 	/**
 	 * Save a game state.
-	 * @param slot	the slot into which the savestate should be stored
-	 * @param desc	a description for the savestate, entered by the user
-	 * @param isAutosave	Expected to be true if an autosave is being created
-	 * @return returns kNoError on success, else an error code.
+	 *
+	 * @param slot        The slot into which the save state should be stored.
+	 * @param desc        Description for the save state, entered by the user.
+	 * @param isAutosave  Expected to be true if an autosave is being created.
+	 *
+	 * @return kNoError on success, otherwise an error code.
 	 */
 	virtual Common::Error saveGameState(int slot, const Common::String &desc, bool isAutosave = false);
 
 	/**
 	 * Save a game state.
-	 * @param stream	The write stream to save the savegame data to
-	 * @param isAutosave	Expected to be true if an autosave is being created
-	 * @return returns kNoError on success, else an error code.
+	 *
+	 * @param stream      The write stream to save the savegame data to.
+	 * @param isAutosave  Expected to be true if an autosave is being created.
+	 *
+	 * @return kNoError on success, otherwise an error code.
 	 */
 	virtual Common::Error saveGameStream(Common::WriteStream *stream, bool isAutosave = false);
 
 	/**
-	 * Indicates whether a game state can be saved.
+	 * Indicate whether a game state can be saved.
 	 */
 	virtual bool canSaveGameStateCurrently();
 
 	/**
-	 * Shows the ScummVM save dialog, allowing users to save their game
+	 * Show the ScummVM save dialog, allowing users to save their game.
 	 */
 	bool saveGameDialog();
 
 	/**
-	 * Shows the ScummVM Restore dialog, allowing users to load a game
+	 * Show the ScummVM Restore dialog, allowing users to load a game.
 	 */
 	bool loadGameDialog();
 
 protected:
 	/**
-	 * Actual implementation of pauseEngine by subclasses. See there
-	 * for details.
+	 * Actual implementation of pauseEngine by subclasses.
+	 *
+	 * @see Engine::pauseEngine
 	 */
 	virtual void pauseEngineIntern(bool pause);
 
-	//@}
+	 /** @} */
 
 
 public:
 
 	/**
-	 * Request the engine to quit. Sends a EVENT_QUIT event to the Event
-	 * Manager.
+	 * Request the engine to quit.
+	 *
+	 * Sends an EVENT_QUIT event to the Event Manager.
 	 */
 	static void quitGame();
 
 	/**
-	 * Return whether the ENGINE should quit respectively should return to the
-	 * launcher.
+	 * Return whether the engine should quit respectively should return to the
+	 * launcher. @todo Doc review, not sure what this is supposed to mean
```

Comment:
Return whether the engine should quit (which means either quitting ScummVM altogether, or returning to the launcher). This should be used in games loops after pulling event to know if we need to exit the game loops.